patsy/eval.py

```diff
@@ -81,6 +82,42 @@ def test_VarLookupDict():
     ds["a"] = 10
     assert ds["a"] == 10
     assert d1["a"] == 1
+    assert ds.get("c") is None
+    assert isinstance(repr(ds), six.string_types)
+
+def ast_names(code):
+    """Iterator that yields all the (ast) names in a Python expression.
+
+    :arg code: A string containing a Python expression.
+    """
+    disallowed_ast_nodes = (ast.Lambda, ast.ListComp, ast.GeneratorExp)
```

Comment:
Let's add a comment here saying something like "# Syntax that allows new name bindings to be introduced is tricky to handle here, so we just refuse to do so"